src/app/toolbar/components/toolbar/toolbar.component.html

```diff
@@ -11,4 +11,8 @@
     
     <button *ngIf="reportDate" mat-button (click)="openDialog()">{{ reportDate | date: 'dd.MM.YYYY' }}</button>
     <button *ngIf="!reportDate" mat-button (click)="openDialog()" i18n>Enter Date</button>
+
+</div>
+<div class="progressbar" *ngIf="progressBarVisible">
+    <mat-progress-bar mode="indeterminate"></mat-progress-bar>
```

Comment:
To be honest we can make determinate progress bar. We have multiple parts, each one saved can increase a progress by one.

Though it will require some bidirectional communication between threads but I think it is possible